pkg/common-controller/groupsnapshot_controller_helper.go

```diff
@@ -599,15 +599,19 @@ func (ctrl *csiSnapshotCommonController) createSnapshotsForGroupSnapshotContent(
 				Labels:     label,
 				Finalizers: []string{utils.VolumeSnapshotInGroupFinalizer},
 			},
-			Spec: crdv1.VolumeSnapshotSpec{
-				Source: crdv1.VolumeSnapshotSource{
-					PersistentVolumeClaimName: &pv.Spec.ClaimRef.Name,
-				},
-			},
+			// The spec stanza is set immediately
 			// The status will be set by VolumeSnapshot reconciler
 		}
 
-		_, err = ctrl.clientset.SnapshotV1().VolumeSnapshotContents().Create(ctx, volumeSnapshotContent, metav1.CreateOptions{})
+		// If we didn't found any persistent volume, we revert back to a static-provisioning
+		// alike object creation
+		if pv != nil {
+			volumeSnapshot.Spec.Source.PersistentVolumeClaimName = &pv.Spec.ClaimRef.Name
+		} else {
+			volumeSnapshot.Spec.Source.VolumeSnapshotContentName = &volumeSnapshotContentName
```

Comment:
I think it should be set to empty string in that case.  We should be consistently treating this as dynamic provisioning.  Setting it to "VolumeSnapshotContentName" indicates it is static provisioning.